### ISSUE TITLE
difference between LabelCrossEntropy and OneHot + SigmoidCrossEntropyWithLogit

### ISSUE THREAD

**Issue by davinwang** [facebookarchive/caffe2#1403]
What's difference between using `LabelCrossEntropy` and decode label with `OneHot` and using `SigmoidCrossEntropyWithLogit`?
I found the loss are different, is that normal?

**Comment by davinwang** [comment]
Let's take MNIST as an example. In `AddTrainingOperators` method, code snippet  
>`    xent = model.LabelCrossEntropy([softmax, label], 'xent')`  
test accuracy are 95%, training curve is as following
![labelcrossentropy](https://user-images.githubusercontent.com/5567847/32209365-e15ad5a4-be42-11e7-95fc-11ddd4717bf9.png)

replace above with
>`    label_int64 = model.Cast(label, None, to=core.DataType.INT64)`
`    SIZE = model.param_init_net.ConstantFill([], 'SIZE', shape=[1,], value=10) # constant`
`    onehot = model.OneHot([label_int64, SIZE], 'onehot')`
`    onehot = model.StopGradient(onehot, onehot)`
`    xent = model.SquaredL2Distance([softmax, onehot], 'xent')`

test accuracy are 90%, training curve listed
![l2](https://user-images.githubusercontent.com/5567847/32209456-35ce4b5c-be43-11e7-95bc-8f66db32fae3.jpg)
and with
>`    label_int64 = model.Cast(label, None, to=core.DataType.INT64)`
`    SIZE = model.param_init_net.ConstantFill([], 'SIZE', shape=[1,], value=10) # constant`
`    onehot = model.OneHot([label_int64, SIZE], 'onehot')`
`    onehot = model.StopGradient(onehot, onehot)`
`    xent = model.SigmoidCrossEntropyWithLogits([softmax, onehot], 'xent')`

The result test accuracy is 32% respectively.
![sigmoidcrossentropywithlogits](https://user-images.githubusercontent.com/5567847/32209890-94886e78-be45-11e7-8e38-53fa27929399.jpg)

**Comment by davinwang** [comment]
Is it normal to have lower accuracy for `SigmoidCrossEntropyWithLogits`? Or did I use it the wrong way?

**Comment by davinwang** [comment]
I kind understand the reason for low accuracy

**Comment by kevinlin311tw** [comment]
Could you please elaborate more about the reason for low accuracy?

**Comment by BIGBALLON** [comment]
@davinwang Could you please elaborate more about the reason for low accuracy and the right way to calculate loss by using one hot label?
